doc/devel/development_setup.rst

```diff
@@ -4,6 +4,51 @@
 Setting up Matplotlib for development
 =====================================
 
+.. note::
+
+    Setting up everything correctly and working with git can be challenging for
+    new contributors. If you have difficulties, feel free to ask for help. We
+    have a dedicated :ref:`contributor-incubator` channel on gitter for
+    onboarding new contributors.
+
+Setting up git and retrieving the latest version of the code
+============================================================
+
+Matplotlib is hosted on `GitHub <https://github.com/matplotlib/matplotlib>`_.
+To contribute you will need to sign up for a `free GitHub account
+<https://github.com/signup/free>`_.
+
+GitHub uses Git for version control, which allows many people to work together
+on the project. See the following links for more information:
+
+- `GitHub help pages <https://help.github.com/>`_
+- `NumPy documentation <https://numpy.org/doc/stable/dev/index.html>`_
+- `pandas documentation <https://pandas.pydata.org/docs/development/contributing.html#working-with-the-code>`_
+
+Forking
+-------
+Go to https://github.com/matplotlib/matplotlib and click the `Fork` button in
+the top right corner to create your own copy of the project.
+
+Now clone your fork to your machine:
+
+.. code-block:: bash
+
+    git clone https://github.com/your-user-name/matplotlib.git
+    cd matplotlib
+    git remote add upstream https://github.com/matplotlib/matplotlib.git
```

Comment:
break the adding upstream into a new section, starting with something like "To keep your local copy of `matplotlib` in sycn with ours, add the offical repository as a remote repository. By convention this repostory is named the upstream. 
then I'd add a step: github has already set up your fork as a remote and named it origin. you can check this using 
`git remote -v`

Reply:
[numpy](https://numpy.org/doc/stable/dev/index.html#development-process-summary) has pulled this apart, [pandas](https://pandas.pydata.org/docs/development/contributing.html#forking) is using one code block.

I've chosen to go with one block because it's looks less verbose and thus less intimidating. YMMV. I like the itemized style of numpy, but it's a little too terse with explanations. :man_shrugging:

Reply:
I don't fine the single block less intimidating, I find it more confusing 'cause it's smashing a lot of steps together.

Reply:
Ok it's single blocks then.